lib/server_connection.ml

```diff
@@ -264,12 +274,7 @@ let rec _next_write_operation t =
   else (
     let reqd = current_reqd_exn t in
     match Reqd.output_state reqd with
-    | Waiting ->
-      (* XXX(dpatti): I don't think we should need to call this, but it is
-         necessary in the case of a streaming, non-chunked body so that you can
-         set the appropriate flag. *)
-      Reqd.flush_response_body reqd;
-      Writer.next t.writer
+    | Waiting -> Writer.next t.writer
```

Comment:
Nb, after my changes I was able to remove this hack, and all the tests still pass. I think the comment was referring to the fact that if you're not using a streaming response, `write_final_if_chunked` starts out true, but then becomes false the first time `transfer_to_writer_with_encoding` is called. Now, the relevant variable doesn't even exist.

Reply:
Sounds good, though I was actually exploring in another branch possibly moving the `Reqd.flush_response_body` outside this entire match statement. The reason for that is because sometimes flushing the body actually changes the `Reqd.output_state`, which means you can get to `Complete` more quickly. Not saying we have to do that here, though I would be curious to see how it works.